internal/app/init/internal/platform/baremetal/baremetal.go

```diff
@@ -67,25 +68,30 @@ func (b *BareMetal) UserData() (data *userdata.UserData, err error) {
 		return data, nil
 	}
 
-	return userdata.Download(option, nil)
+	return userdata.Download(*option, nil)
 }
 
 // Prepare implements the platform.Platform interface.
 func (b *BareMetal) Prepare(data *userdata.UserData) (err error) {
 	return install.Prepare(data)
 }
 
-// Install provides the functionality to install talos by
-// download the necessary bits and write them to a target device
-// nolint: gocyclo, dupl
+// Install provides the functionality to install talos by downloading the
+// necessary bits and write them to a target device
+// nolint: dupl
 func (b *BareMetal) Install(data *userdata.UserData) (err error) {
-	var cmdlineBytes []byte
-	cmdlineBytes, err = kernel.ReadProcCmdline()
-	if err != nil {
-		return err
-	}
-	if err = install.Install(string(cmdlineBytes), data); err != nil {
-		return errors.Wrap(err, "failed to install to bare metal")
+	cmdline := kernel.NewCmdline(kspp.RequiredKSPPKernelParametersString)
```

Comment:
This is used in three different places, we should probably make this something like `kernel.DefaultParameters()`